app/models/user.rb

```diff
@@ -34,6 +36,19 @@ def valid_otp_attempt?(otp_attempt)
     validate_and_consume_otp!(otp_attempt)
   end
 
+  def send_otp(session_mobile_number)
+    if send_otp_by_sms?
+      number = mobile_number.presence || session_mobile_number
+      TwoFactor::SmsNotification.new(number, current_otp).deliver!
+    else
+      UserMailer.otp_mail(self).deliver_now
+    end
+  end
+
+  def send_otp_by_sms?
+    otp_delivery_method == "sms"
```

Comment:
does doing just `otp_delivery_method.sms?` work here ?

Reply:
I'd thought that...but `otp_delivery_method` just returns a string. As far as I can see a lot of the enum magic only happens if the column is `status` 😞

Reply:
Ah is that so! Good to know